### PR TITLE
Create quiz

### PR DESCRIPTION
**Pull request by AdriaenXO**
Added support for creating a new quiz and adding questions to it.
Needs: error handling (will be added once this is approved => using toasts)
Needs: testing

### CODE REVIEW

File: client/src/components/CreateQuestion.vue [hunk 1]
```diff
@@ -0,0 +1,142 @@
+<template>
+  <div>
+    <div class="container">
+      <h2>Create question</h2>
+      <input class="input-textarea" type="text" placeholder="Question" v-model="text" />
+      <ul id="answers">
```
Comment (BartoszGamza) [inline client/src/components/CreateQuestion.vue:6]: why id?

File: client/src/components/CreateQuestion.vue [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        <li v-for="answer in answers" :key="answer.localId">
+          <input type="checkbox" :value="answer.id" v-model="correctAnswers" />
+          <input class="input-textarea" type="text" placeholder="Answer" v-model="answer.text" />
+        </li>
+      </ul>
+      <button class="signup-form-button active-theme" v-on:click.prevent="addNewAnswer">Add Answer</button>
```
Comment (BartoszGamza) [inline client/src/components/CreateQuestion.vue:12]: `v-on = @` fyi 👍

File: client/src/components/CreateQuestion.vue [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      <button
+        class="signup-form-button active-theme"
+        v-on:click.prevent="submitQuestion"
```
Comment (BartoszGamza) [inline client/src/components/CreateQuestion.vue:15]: are the prevent's here necessary? if its not a button of `submit` type or even form there shouldn't be need, also it shouldn't have any effect

File: client/src/components/CreateQuestion.vue [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+</template>
+
+<script>
+import { mapActions } from 'vuex'
+export default {
+  name: 'CreateQuestion',
+  data () {
+    return {
+      text: '',
+      answers: [
```
Comment (BartoszGamza) [inline client/src/components/CreateQuestion.vue:29]: so, the number of answers is fixed to 4?

File: client/src/components/CreateQuestion.vue [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        this.correctAnswers = []
+      } else {
+        console.log(resp)
+      }
+    },
+    addNewAnswer () {
+      let localId = 0
+      this.answers.forEach(answer => {
+        if (answer.id > localId) localId = answer.id
+      })
```
Comment (BartoszGamza) [inline client/src/components/CreateQuestion.vue:86]: this could just be `this.answers[this.answers.length -1]` id

File: client/src/router.js [hunk 6]
```diff
@@ -24,7 +25,7 @@ export default new Router({
       name: 'signup',
       component: SignUpPage,
       beforeEnter (to, from, next) {
-        if (store.getters.isAuthenticated) {
+        if (store.getters['auth/isAuthenticated']) {
```
Comment (BartoszGamza) [inline client/src/router.js:28]: gr8 I forgot this 🙏

File: client/src/store/modules/quiz.js [hunk 7]
```diff
@@ -22,6 +22,20 @@ export const quiz = {
           .catch(() => null)
         if (questions) commit('setQuizQuestions', questions)
       }
+    },
+    addQuiz ({ commit }, quiz) {
```
Comment (BartoszGamza) [inline client/src/store/modules/quiz.js:26]: if any of those does not have effect for the store it should not be in the store imo, could you move those functions to some helpers file? Actions by its origin modify the state and do not return any value

File: client/src/components/CreateQuestion.vue (hunk 1 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:6]: Could be omitted but does not hurt I guess?

File: client/src/components/CreateQuestion.vue (hunk 4 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:29]: No, there is a button that adds more of them; 4 are the initial state

File: client/src/components/CreateQuestion.vue (hunk 5 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:86]: Theoretically yes, but I had in mind an idea to be able to delete answers in the future but we dont have time for that

**Comment by BartoszGamza** [comment]
Sorry if I sound too picky but there are key rules of Vue that we could respect rather than get surpsied later - as I did, thats why I made such huge refactor yesterday

File: client/src/components/CreateQuestion.vue (hunk 5 above)
Comment (BartoszGamza) [reply client/src/components/CreateQuestion.vue:86]: if number stayed at 1 at minium there fine, besides the id could be added on request phase - you dont need it for rendering

File: client/src/components/CreateQuestion.vue (hunk 4 above)
Comment (BartoszGamza) [reply client/src/components/CreateQuestion.vue:29]: 2 sounds more reasonable if you cant remove

File: client/src/components/CreateQuestion.vue (hunk 2 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:12]: Yes, I personally prefer `v-on`

File: client/src/components/CreateQuestion.vue (hunk 1 above)
Comment (BartoszGamza) [reply client/src/components/CreateQuestion.vue:6]: no it doesn't hurt but doesn't have sense either, I was asking if it actually serves a purpose or you forgot to remove after some change

File: client/src/components/CreateQuestion.vue (hunk 3 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:15]: Resolved

File: client/src/components/CreateQuestion.vue (hunk 1 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:6]: It was just used for styling

File: client/src/components/CreateQuestion.vue (hunk 5 above)
Comment (AdriaenXO) [reply client/src/components/CreateQuestion.vue:86]: Reduced number to 2 in the upcoming commit

File: client/src/store/modules/quiz.js (hunk 7 above)
Comment (AdriaenXO) [reply client/src/store/modules/quiz.js:26]: Moved to another file in an upcoming commit

**Pull request merged**